cmd/skaffold/app/cmd/runner.go

```diff
@@ -79,7 +79,7 @@ func runContext(out io.Writer, opts config.SkaffoldOptions) (*runcontext.RunCont
 	}
 	setDefaultDeployer(cfgSet)
 
-	if err := validation.Process(cfgSet); err != nil {
+	if err := validation.Process(cfgSet, validation.DefaultConfig); err != nil {
```

Comment:
I think instead of using `validation.DefaultConfig` everywhere, we can decide based on the command executed. Commands like `inspect`, `diagnose`, `fix` and also `build` can skip while the commands like `deploy`, `render`, `dev`, `run`, and `debug` should do the validation.

So something like:
```go
func getValidationOpts(opts config.SkaffoldOptions) validation.Options {
    switch opts.Mode(){
        case config.RunModes.Dev, config.RunModes.Deploy, config.RunModes.Run, config.RunModes.Debug, config.RunModes.Render:
            return validation.Options{CheckDeploySource: true}
        default:
            return validation.Options{}
    }
}
```

Reply:
Done